tests/test_telemetry_monitor.py

```diff
@@ -0,0 +1,1142 @@
+# BSD 2-Clause License
```

Comment:
Several tests in this file require the TM to be active in order for them to not pass. Therefore, if a tester were to explicitly `export SMARTSIM_FLAG_TELEMETRY=0` they will get spurious errors.

By my estimate these tests are:
- `test_telemetry_single_model`
- `test_telemetry_single_model_nonblocking`
- `test_telemetry_serial_models`
- `test_telemetry_serial_models_nonblocking`
- `test_telemetry_db_only_with_generate`
- `test_telemetry_db_only_without_generate`
- `test_telemetry_db_and_model`
- `test_telemetry_ensemble`
- `test_telemetry_colo`
- `test_telemetry_autoshutdown` (all perms)
- `test_multistart_experiment` (all perms)

Maybe add a:
```py
@pytest.fixture(autouse=True)
def turn_on_tm(monkeypatch):
    import smartsim._core.config.config
    monkeypatch.setattr(
        smartsim._core.config.config.Config,
        "telemetry_enabled",
        property(lambda self: True))
    yield
```
to the top of this module?